File: 프로그래머스/unrated/120924. 다음에 올 숫자/다음에 올 숫자.js
function nextNumber(common) {
    let next = 0;
    const lastIndex = common.length - 1;
    
    if (common[1] - common[0] === common[2] - common[1]) {
        next = common[lastIndex] * 2 - common[lastIndex-1];
    } else {
        next = common[lastIndex] ** 2 / common[lastIndex-1];
    }
    
    return next;
}

function solution(common) {
    let answer = 0;
    
    answer = nextNumber(common);
    
    return answer;
}

File: 프로그래머스/0/120924. 다음에 올 숫자/다음에 올 숫자.py
def solution(common):
    answer = 0
    d = common[1] - common[0] 
    if d == (common[2] - common[1]):
        answer = common[-1] + d
    else:
        r = common[1] / common[0]
        answer = common[-1] * r
    return answer